### PR TITLE
[FIXED JENKINS-42390] Search results were not correctly encoding URL query parameters

### PR DESCRIPTION
**Pull request by stephenc**
See [JENKINS-42390](https://issues.jenkins-ci.org/browse/JENKINS-42390)

@reviewbybees

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
🐝

**Review by daniel-beck — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
@reviewbybees done assuming that @daniel-beck works for CloudBees.
Let's wait for the community feedback for a while

File: core/src/main/resources/hudson/search/Search/search-failed.jelly [hunk 1]
```diff
@@ -43,13 +43,13 @@ THE SOFTWARE.
           <ol>
             <j:forEach var="i" items="${items}">
               <li id="item_${i.path}">
-                <a href="?q=${i.path}">${i.path}</a>
+                <a href="?q=${it.encodeQuery(i.path)}">${i.path}</a>
```
Comment (jglick) [inline core/src/main/resources/hudson/search/Search/search-failed.jelly]: Probably reproducible in a `WebClient`-based test if you search for anchor tags.

**Review by jglick — APPROVED** [review]

File: core/src/main/resources/hudson/search/Search/search-failed.jelly (hunk 1 above)
Comment (stephenc) [inline core/src/main/resources/hudson/search/Search/search-failed.jelly]: The problem is creating search results that need encoding without depending on plugins

**Pull request merged**